src/script/design/Element.ts

```diff
@@ -16,8 +18,11 @@ export class ElementView {
     }
     public static toggleVisibility(domObj: HTMLElement): HTMLElement {
         domObj.style.display = domObj.style.display === "none" ? "inline-block" : "none";
+        const url = document.location.search + document.location.hash;
+        HistoryManager.UpdateHistory(url, Main.viModel);
```

Comment:
Why do we need to update history here? And also, does this function call actually work? The `if` condition for the `document.referrer` will end up not updating history when called from toggleVisibility(), won't it?

Reply:
1.We need the UpdateHistory here to get the right sourcemodel after the case toggles.
2. Yes, this function works to toggle the case in multiframe and i saw a reference of it in TabControls too. So I believe this would solve both the problems.
3.It will replace the history and also update the session storage. The session storage wont get updated only if it is an subVi(in  `documet.refferer != "http://127.0.0.1:8081/results.html"`).

Reply:
If sessionStorage.set() gets called just before we "leave" a VI and navigate to the next one, it should be enough, right? Why do we need to update the sessionStorage variable every single time we make a change to the SourceModel?

Reply:
Since the original idea of saving state only once in LoadFile() does not seem to work, can you just add a TODO here with my name to investigate it?

Reply:
Okay